src/sql/workbench/contrib/query/browser/keyboardQueryActions.ts

```diff
@@ -206,6 +206,29 @@ export class CopyQueryWithResultsKeyboardAction extends Action {
 	}
 }
 
+export class EstimatedExecutionPlanKeyboardAction extends Action {
+	public static ID = 'estimatedExecutionPlanKeyboardAction';
+	public static LABEL = nls.localize('estimatedExecutionPlanKeyboardAction', "Display Estimated Execution Plan");
+
+	constructor(
+		id: string,
+		label: string,
+		@IEditorService private _editorService: IEditorService
+	) {
+		super(id, label);
+		this.enabled = true;
+	}
+
+	public override run(): Promise<void> {
+		const editor = this._editorService.activeEditorPane;
+		if (editor instanceof QueryEditor) {
+			editor.input.runQuery(editor.getSelection(), { displayEstimatedQueryPlan: true });
+		}
+
+		return Promise.resolve(undefined);
```

Comment:
Oh actually I wasn't even looking at this closely enough - you shouldn't be returning anything at all here. I'd just make this function async and then remove the return entirely (it'll automatically default to returning a void promise result)